src/rest_api.py

```diff
@@ -44,6 +53,55 @@ def liveness():
     return flask.jsonify({}), 200
 
 
+@app.route('/api/v1/sync-graph-data', methods=['POST'])
+@login_required
+def sync_data():
+    """
+    Endpoint for carrying a sync of graph db data.
+
+    Takes in value to either call sync of non cve version
+    or latest version or both.
+
+    valid input: {
+        "non_cve_sync": True / False,
+        "latest_version_sync": True / false,
+        "cve_ecosystem": ['maven', 'pypi, 'npm']
+    }
+
+    """
+    resp = {
+        "success": True,
+        "message": "Sync operation started ->"
+    }
+
+    input_json = request.get_json()
+    logger.info("sync-graph-data called with the input {i}".format(i=input_json))
+    non_cve_sync = input_json.get('non_cve_sync', False)
+    if non_cve_sync == "true":
+        cve_ecosystem = input_json.get('cve_ecosystem', [])
+        if len(cve_ecosystem) == 0:
+            resp['success'] = False
+            resp['message'] = "Incorrect data.. Send cve_ecosystem for non_cve_sync operation"
+            logger.error("Incorrect data.. Send cve_ecosystem for non_cve_sync operation")
+            return flask.jsonify(resp), 400
+        url = "http://{host}:{port}/{endpoint}".format(host=_SERVICE_HOST,
+                                                       port=_SERVICE_PORT,
+                                                       endpoint=_CVE_SYNC_ENDPOINT)
+        logger.info("Calling non cve sync with {i}".format(i=cve_ecosystem))
+        _session.post(url, json=cve_ecosystem)
+        resp['message'] = resp['message'] + " for non cve version"
+    latest_version_sync = input_json.get('latest_version_sync', False)
+    if latest_version_sync == "true":
```

Comment:
@yzainee as per the swagger spec, `'latest_version_sync'` is a boolean field which should not be compared with a string `"true"`.

The boolean value comparison in Python happens like `if latest_version_sync` or like `if not latest_version_sync `.